packages/eslint-plugin/src/rules/no-unused-vars.ts

```diff
@@ -197,15 +197,20 @@ export default util.createRule<Options, MessageIds>({
           return false;
         });
         if (isOnlySelfReferenced) {
-          context.report({
-            node: variable.identifiers[0],
-            messageId: 'unusedVar',
-            data: {
-              varName: variable.name,
-              action: 'defined',
-              additional: '',
-            },
-          });
+          if (
+            varsIgnorePattern &&
+            varsIgnorePattern.toString() !== variable.name
```

Comment:
`varsIgnorePattern` is a regexp (or perhaps still a regexp string).

See usage in eslint:
`if (config.varsIgnorePattern && config.varsIgnorePattern.test(def.name.name)) {`

https://github.com/eslint/eslint/blob/551b1e9fbc463db4869e738a8228868f7323788c/lib/rules/no-unused-vars.js#L590

Reply:
And this error should be reported also when no `varsIgnorePattern` is given.

Reply:
Do we need a similar change in line 148?

Reply:
Initially, I was confused on the same thing. 
But, I found this as my reference to go for this method:
```ts
var varIgnorePattern: "all" | "local" | {
  vars?: "all" | "local" | undefined;
  varIgnorePattern:?: string | undefined;
  args: "all" | "after-used" | "none" | undefined;
  .........
  caughtErrorsIgnorePattern?: string | undefined;
```

@bradzacher can help us in this.

Reply:
Doing the regex, I am having an error
```ts
varIgnorePattern.test(variable.name)
```

Reply:
Hy @HolgerJeromin, Can you guide me ?

Reply:
No, sorry. I never hacked around around eslint

Reply:
So here you'll need to convert the string to a regexp:

```
const varIgnorePattern =
  typeof options === 'object' && options.varIgnorePattern != null
    ? new RegExp(options.varsIgnorePattern, "u")
    : null;
```

Reply:
I have worked with the regex but how can I improve codecov ? @bradzacher

Reply:
you'll need some tests! right now you have none at all.